core/node/proof_data_handler/src/errors.rs

```diff
@@ -19,6 +20,13 @@ impl From<DalError> for RequestProcessorError {
 impl IntoResponse for RequestProcessorError {
     fn into_response(self) -> Response {
         let (status_code, message) = match self {
+            RequestProcessorError::GeneralError(err) => {
+                tracing::error!("Error: {:?}", err);
+                (
+                    StatusCode::BAD_GATEWAY,
```

Comment:
Bikeshedding: "Bad gateway" [doesn't look like](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502) the best status code here:

> indicates that a server was acting as a gateway or [proxy](https://developer.mozilla.org/en-US/docs/Glossary/Proxy_server) and that it received an invalid response from the upstream server.

IMO, [500 Internal Server Error](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500) would work better.

Reply:
good catch!

Reply:
fixed